docs/get-started/using-layers.md

```diff
@@ -1,49 +1,125 @@
+They should not be used in new applications as they will be removed in future
+deck.gl releases.
+
+### Sample Layers
+
+Deck.gl provides a number of sample layers in the examples folders intended to
+illustrate various ideas and approaches to how layers can be designed. These
+layers sometimes have documentation in the example code, but they are not
+listed here in the official documentation.
+
+## Layer Creation, Update and Destruction
 
 Every time some state in your application that affects visualization changes,
 you simply create new layer instances with updated properties and pass them to
 deck.gl for rendering.
 
+After matching your newly supplied layers with the layers you provided
+in your previous call to deck.gl, the state of any old layers is copied to
+the new layers, and the old layers are discarded by deck.gl.
+
+The application does not have to be aware about this, as long as it keeps
+rendering new layers with the same `id` they will be matched and the existing
+state of that layer will be updated accordingly
+
 The constant creation and disposal of layer instances may seem wasteful,
 however the creation and recycling of JavaScript objects is quite efficient
-in modern JavaScript environments.
+in modern JavaScript environments, and this is very similar to how React
+works where every render cycle generates a new tree of ReactElement instances,
+so the model is proven.
 
-The advantage of this model is that it enables a functional, reactive
-application model, where the UI is completely re-rendered to correspond to
-application state changes. There is no need to distribute your application
-state throughout your UI components.
+The layer lifecycle is documented in detail in the section about writing layers.
 
+## Notable Layer Properties
 
-## The id Property
+### The id Property
 
-The `id` property is similar to the `key` property on React components. It is
-used to ensure that new components are matched with their counterparts from the
-previous rendering cycle. However, unlike React.js, the `id` property is not
-optional in deck.gl. It must be unique for each layer or deck.gl will fail to
-match layers.
+It is used to ensure that new components are matched with their counterparts
+from the previous rendering cycle. the `id` property defaults to the layer name
+so if you only have a single layer of each type you may not need to supply it.
 
+However as soon as you add multiple instances of a specific layer (say
+two `ScatterplotLayer`s) their `id`s must be unique for each layer or deck.gl
+will fail to match layers.
 
-## The data Property and Accessors
+The `id` property is similar to the `key` property on React components.
+However deck.gl relies on this property even more than React does.
+
+
+```
+  <DeckGL layers={[
+    new ScatterplotLayer({id: 'big-points', data: ..., ...}),
+    new ScatterplotLayer({data: 'small-points', data: ..., ...}),
+  ]}/>
+```
+
+
+### The data Property and Accessors
 
```

Comment:
The data and accessor has been discussed in the "rendering a single layer" section (which is where it should be discussed). so it seems repetitive here